animation_nodes/nodes/mesh/set_vertex_weights.py

```diff
@@ -0,0 +1,145 @@
+                ("Index", "index"), ("Indices", "indices")))
+            self.newInput(VectorizedSocket("Float", "useFloatList",
+                ("Weight", "weight"), ("Weights", "weights")))
+        elif self.mode == "ALL":
+            self.newInput(VectorizedSocket("Float", "useFloatList",
+                ("Weight", "weight"), ("Weights", "weights")))
+        
+        self.newOutput("Object", "Object", "object")
+
+    def draw(self, layout):
+        layout.prop(self, "mode", text = "")
+
+    def drawAdvanced(self, layout):
+        layout.prop(self, "groupIdentifierType", text = "Type")
+
+    def getExecutionFunctionName(self):
+        if self.mode == "INDEX":
+            if self.useIndexList:
+                if self.useFloatList:
+                    return "execute_IndicesList"
+                else:
+                    return "execute_IndicesSingle"
+            else:
+                return "execute_Index"
+        elif self.mode == "ALL":
+            if self.useFloatList:
+                return "execute_AllList"
+            else:
+                return "execute_AllSingle"
+
+    def execute_Index(self, object, identifier, index, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        vertexGroup.add([index], weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesSingle(self, object, identifier, indices, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        vertexGroup.add(indices, weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesList(self, object, identifier, indices, weights):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        for i, index in enumerate(indices):
+            vertexGroup.add([index], weights[i], "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_AllSingle(self, object, identifier, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        indices = list(range(0, len(object.data.vertices)))
```

Comment:
The zero is redundant here.

Reply:
Removed the zero from the _range_ function.